samples/batch/batch-ingestion-client/Connector/BatchClient.cs

```diff
@@ -88,127 +88,121 @@ public static async Task<Uri> PostTranscriptionAsync(TranscriptionDefinition tra
             }
         }
 
         private static async Task DeleteAsync(string path, string subscriptionKey, TimeSpan timeout, ILogger log)
         {
-            using (var requestMessage = new HttpRequestMessage(HttpMethod.Delete, path))
+            using var requestMessage = new HttpRequestMessage(HttpMethod.Delete, path);
+            if (!string.IsNullOrEmpty(subscriptionKey))
+            {
+                requestMessage.Headers.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
+            }
+
+            try
             {
-                if (!string.IsNullOrEmpty(subscriptionKey))
+                using var cts = new CancellationTokenSource();
+                cts.CancelAfter(timeout);
+                var responseMessage = await HttpClient.SendAsync(requestMessage, cts.Token).ConfigureAwait(false);
+
+                if (IsThrottledOrTimeoutStatusCode(responseMessage.StatusCode))
                 {
-                    requestMessage.Headers.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
+                    throw new TimeoutException(responseMessage.StatusCode.ToString());
                 }
 
-                try
+                if (!responseMessage.IsSuccessStatusCode)
                 {
-                    using var cts = new CancellationTokenSource();
-                    cts.CancelAfter(timeout);
-                    var responseMessage = await HttpClient.SendAsync(requestMessage, cts.Token).ConfigureAwait(false);
+                    var failureMessage = $"Failure - Status Code: {responseMessage.StatusCode}";
 
-                    if (IsThrottledOrTimeoutStatusCode(responseMessage.StatusCode))
+                    if (responseMessage.Content != null)
                     {
-                        throw new TimeoutException(responseMessage.StatusCode.ToString());
+                        var body = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+                        failureMessage += $", Body: {body}";
                     }
 
-                    if (!responseMessage.IsSuccessStatusCode)
-                    {
-                        var failureMessage = $"Failure - Status Code: {responseMessage.StatusCode}";
-
-                        if (responseMessage.Content != null)
-                        {
-                            var body = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
-                            failureMessage += $", Body: {body}";
-                        }
-
-                        log.LogInformation(failureMessage);
-                        throw new WebException(failureMessage);
-                    }
-                }
-                catch (OperationCanceledException)
-                {
-                    throw new TimeoutException($"The operation has timed out after {timeout.TotalSeconds} seconds.");
+                    log.LogInformation(failureMessage);
+                    throw new WebException(failureMessage);
```

Comment:
The error handling code looks similar, maybe consider to have a helper function to avoid duplication.

Reply:
makes sense, using a shared helper now!